### PR TITLE
Modus Autocomplete: Disable brower's autocomplete on text input

### PR DESCRIPTION
**Pull request by yohernandez**
## Description

Adding a new property to set the autocomplete attribute for text inputs and the autocomplete input.
The property follows https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete.
Adding e2e tests to ensure changes on the property are honored.

References #1985 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

Tested manually and in Storybook. Added e2e tests.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [X] I have checked my code and corrected any misspellings

### CODE REVIEW

File: stencil-workspace/src/components/modus-autocomplete/modus-autocomplete.tsx [hunk 1]
```diff
@@ -32,6 +32,9 @@ export class ModusAutocomplete {
   /** The autocomplete's aria label. */
   @Prop() ariaLabel: string | null;
 
+  /** (optional) Sets autocomplete. */
+  @Prop() autocomplete: string | null;
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-autocomplete/modus-autocomplete.tsx:36]: adding new property autocomplete

File: stencil-workspace/src/components/modus-text-input/modus-text-input.e2e.ts [hunk 2]
```diff
@@ -266,4 +266,17 @@ describe('modus-text-input', () => {
     await page.waitForChanges();
     expect(valueChange).toHaveReceivedEvent();
   });
+
+  it('ensures it renders changes to autocomplete', async () => {
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-text-input/modus-text-input.e2e.ts:270]: adding new test to ensure the attribute is passed to the inner input element

File: stencil-workspace/src/components/modus-text-input/modus-text-input.tsx [hunk 3]
```diff
@@ -14,6 +14,9 @@ export class ModusTextInput {
   /** (optional) The input's aria-label. */
   @Prop() ariaLabel: string | null;
 
+  /** (optional) Sets autocomplete on the input. */
+  @Prop() autocomplete: string | null;
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-text-input/modus-text-input.tsx:18]: adding new property autocomplete

File: stencil-workspace/storybook/stories/components/modus-autocomplete/modus-autocomplete-storybook-docs.mdx [hunk 4]
```diff
@@ -91,6 +91,7 @@ interface ModusAutocompleteOption {
 | Property                    | Attribute                       | Description                                       | Type                                    | Default                                       |
 | --------------------------- | ------------------------------- | ------------------------------------------------- | --------------------------------------- | --------------------------------------------- |
 | `ariaLabel`                 | `aria-label`                    | The autocomplete's aria label.                    | `string`                                | `undefined`                                   |
+| `autocomplete`              | `autocomplete`                  | The autocomplete's autocomplete.                  | `string`                                | `undefined`                                   |
```
Comment (yohernandez) [inline stencil-workspace/storybook/stories/components/modus-autocomplete/modus-autocomplete-storybook-docs.mdx:94]: updating Storybook

File: stencil-workspace/storybook/stories/components/modus-autocomplete/modus-autocomplete.stories.tsx [hunk 5]
```diff
@@ -12,6 +12,13 @@ export default {
         type: { summary: 'string' },
       },
     },
+    autocomplete: {
```
Comment (yohernandez) [inline stencil-workspace/storybook/stories/components/modus-autocomplete/modus-autocomplete.stories.tsx:15]: updating Storybook's stories

File: stencil-workspace/storybook/stories/components/modus-text-input/modus-text-input.stories.tsx [hunk 6]
```diff
@@ -12,6 +12,13 @@ export default {
         type: { summary: 'string' },
       },
     },
+    autocomplete: {
```
Comment (yohernandez) [inline stencil-workspace/storybook/stories/components/modus-text-input/modus-text-input.stories.tsx:15]: updating Storybook's stories

File: stencil-workspace/storybook/stories/components/modus-autocomplete/modus-autocomplete-storybook-docs.mdx (hunk 4 above)
Comment (rthanga1) [reply stencil-workspace/storybook/stories/components/modus-autocomplete/modus-autocomplete-storybook-docs.mdx:94]: Can you explain what `The autocomplete's autocomplete` mean? What does it do when we enable this property?

Comment (yohernandez) [reply stencil-workspace/storybook/stories/components/modus-autocomplete/modus-autocomplete-storybook-docs.mdx:94]: @rthanga1 We want to disable the browser autocomplete/autofill behavior in the input element inside the modus-autocomplete. 
image

If a value is provided for this property then the value is propagated to the modus-text-input component inside the modus-autocomplete and then to the input element inside the modus-text-input

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete

**Comment by cjwinsor** [comment]
Remove `autocomplete` property from `modus-autocomplete`.  Keep it on `modus-text-input` and default to `off` from inside the `modus-autocomplete` component.  If later on there is a use case to enable autocomplete for this `modus-autocomplete` then we can address how to update `modus-autocomplete`

File: stencil-workspace/src/components/modus-autocomplete/modus-autocomplete.spec.ts [hunk 7]
```diff
@@ -11,7 +11,7 @@ describe('modus-autocomplete', () => {
     <modus-autocomplete>
       <mock:shadow-root>
         <div class="autocomplete medium">
-          <modus-text-input includesearchicon="" size="medium"></modus-text-input>
+          <modus-text-input autocomplete="off" includesearchicon="" size="medium"></modus-text-input>
```
Comment (dnarey) [inline stencil-workspace/src/components/modus-autocomplete/modus-autocomplete.spec.ts:14]: Default text input's autocomplete property to off

**Comment by dnarey** [comment]
> Remove `autocomplete` property from `modus-autocomplete`. Keep it on `modus-text-input` and default to `off` from inside the `modus-autocomplete` component. If later on there is a use case to enable autocomplete for this `modus-autocomplete` then we can address how to update `modus-autocomplete`

These changes have been completed.  Thanks!

**Review by cjwinsor — APPROVED** [review]

**Pull request merged**